tests/includes/SiteConfigurationTest.php

```diff
@@ -0,0 +1,416 @@
+<?php
+/******************************************************************************
+ * Wikipedia Account Creation Assistance tool                                 *
+ *                                                                            *
+ * All code in this file is released into the public domain by the ACC        *
+ * Development Team. Please see team.json for a list of contributors.         *
+ ******************************************************************************/
+
+namespace Waca\Tests;
+
+use Waca\SiteConfiguration;
+
+class SiteConfigurationTest extends \PHPUnit_Framework_TestCase
+{
+	private $si;
+
+	function setUp() {
+		$this->si = new SiteConfiguration();
+	}
+
+	function tearDown() {
+		unset($this->si);
+	}
+
+	function testSetUp() {
+		$this->assertInstanceOf('\Waca\SiteConfiguration', $this->si);
+	}
+
+	function testBaseUrl() {
+		$newValue = "http://localhost/testAwesome/";
+
+		$this->assertEquals($this->si->getBaseUrl(), null);
+
+		$this->assertInstanceOf('Waca\SiteConfiguration', $this->si->setBaseUrl($newValue));
```

Comment:
It'd be better to just use `SiteConfiguration::class` here, so we don't have a bunch of magic strings that will break unexpectedly if we refactor things.